### PR TITLE
Initial commit of Google Gemini LLM service.

### PR DESCRIPTION
**Pull request by kwindla**
Gemini text input works. We translate from OpenAILLMContext format on the fly in the GoogleLLMService implementation.

This commit also implements image input (vision) in both the GoogleLLMService and in the OpenAILLMService. Image input is a hack and needs to be revisited. OpenAI expects images to be uploaded as base64-encoded JPEGs. Google does not require the base64 encoding. Other than for images, we use the OpenAI format as our standard, but base64-encoding the images and then unencoding them in the GoogleLLMService feels wasteful.

### CODE REVIEW

File: macos-py3.10-requirements.txt [hunk 1]
```diff
@@ -1,32 +1,33 @@
+WARNING: --strip-extras is becoming the default in version 8.0.0. To silence this warning, either use --strip-extras to opt into the new default or use --no-strip-extras to retain the existing behavior.
 #
-# This file is autogenerated by pip-compile with Python 3.10
+# This file is autogenerated by pip-compile with Python 3.11
```
Comment (aconchillo) [inline macos-py3.10-requirements.txt:3]: nit: I try to generate this with python 3.10 just to make sure it works there. on macos i do:

```
brew install python@3.10
python3.10 -m venv venv
...
...
pip-compile --all-extras pyproject.toml
```

File: src/pipecat/services/openai.py [hunk 2]
```diff
@@ -65,8 +67,21 @@ async def _stream_chat_completions(
...
+                text = message["content"]
+                message["content"] = [
+                    {"type": "text", "text": text},
+                    {"type": "image_url", "image_url": {"url": f"data:image/jpeg;base64,{encoded_image}"}}
+                ]
+                del message["data"]
+                del message["mime_type"]
+
+        # messages_for_log = json.dumps(messages)
+        # logger.debug(f"Generating chat: {messages_for_log}")
```
Comment (aconchillo) [inline src/pipecat/services/openai.py:84]: nit: remove or re-add?

**Review by aconchillo — APPROVED** [review]

**Review by chadbailey59 — APPROVED** [review]
LGTM 👍

File: macos-py3.10-requirements.txt (hunk 1 above)
Comment (kwindla) [reply macos-py3.10-requirements.txt]: fixed.

**Pull request merged**